pkg/scm/gitlab/client_actioner_assign_test.go

```diff
@@ -0,0 +1,175 @@
+package gitlab_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/jippi/scm-engine/pkg/config"
+	"github.com/jippi/scm-engine/pkg/scm"
+	"github.com/jippi/scm-engine/pkg/scm/gitlab"
+	"github.com/jippi/scm-engine/pkg/state"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type evalContextMock struct {
+	mock.Mock
+}
+
+func (c *evalContextMock) IsValid() bool {
+	return c != nil
+}
+
+func (c *evalContextMock) SetWebhookEvent(in any) {
+	c.Called(in)
+}
+
+func (c *evalContextMock) SetContext(ctx context.Context) {
+	c.Called(ctx)
+}
+
+func (c *evalContextMock) GetDescription() string {
+	args := c.Called()
+
+	return args.String(0)
+}
+
+func (c *evalContextMock) CanUseConfigurationFileFromChangeRequest(ctx context.Context) bool {
+	args := c.Called(ctx)
+
+	return args.Bool(0)
+}
+
+func (c *evalContextMock) AllowPipelineFailure(ctx context.Context) bool {
+	args := c.Called(ctx)
+
+	return args.Bool(0)
+}
+
+func (c *evalContextMock) TrackActionGroupExecution(group string) {
+	c.Called(group)
+}
+
+func (c *evalContextMock) HasExecutedActionGroup(group string) bool {
+	args := c.Called(group)
+
+	return args.Bool(0)
+}
+
+func (c *evalContextMock) GetCodeOwners() scm.Actors {
+	args := c.Called()
+
+	if actors, ok := args.Get(0).(scm.Actors); ok {
+		return actors
+	}
+
+	return nil
+}
+
+func (c *evalContextMock) GetReviewers() scm.Actors {
+	args := c.Called()
+
+	if actors, ok := args.Get(0).(scm.Actors); ok {
+		return actors
+	}
+
+	return nil
+}
+
+func TestAssignReviewers(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                      string
+		step                      config.ActionStep
+		mockGetReviewersResponse  scm.Actors
+		mockGetCodeOwnersResponse scm.Actors
+		wantUpdate                *scm.UpdateMergeRequestOptions
+		wantErr                   error
+	}{
+		{
+			name: "should error on no source provided",
```

Comment:
should we default to `codeowners` as source, seems like a safe default (and currently the only valid value)